src/test/java/org/devcon/ticket/UseTicketTest.java

```diff
@@ -150,19 +150,59 @@ public void testDecoding() throws InvalidObjectException {
+    IdentifierAttestation att = HelperTest.makeUnsignedStandardAtt(subjectKeys.getPublic(), ATTESTATION_SECRET, MAIL, 15); //valid for 15 seconds
     SignedIdentifierAttestation signed = new SignedIdentifierAttestation(att, attestorKeys);
     Ticket ticket = new Ticket(MAIL, CONFERENCE_ID, TICKET_ID, TICKET_CLASS, ticketIssuerKeys, TICKET_SECRET);
     AttestedObject<Ticket> useTicket = new AttestedObject<>(ticket, signed, ATTESTATION_SECRET, TICKET_SECRET, UN, crypto);
-
+    
     //now attempt to dump data from contract:
     TicketAttestationReturn tar = contract.callVerifyTicketAttestation(useTicket.getDerEncoding());
 
+    System.out.println("Call SmartContract to check Ticket");
+
     //check returned values
     assertTrue(tar.subjectAddress.equalsIgnoreCase(SignatureUtility.addressFromKey(subjectKeys.getPublic())));
     assertTrue(tar.issuerAddress.equalsIgnoreCase(SignatureUtility.addressFromKey(ticketIssuerKeys.getPublic())));
     assertTrue(tar.attestorAddress.equalsIgnoreCase(SignatureUtility.addressFromKey(attestorKeys.getPublic())));
+    assertTrue(tar.timeStampValid);
     assertEquals(Numeric.toBigInt(tar.ticketId), TICKET_ID);
+
+    System.out.println("Test passed");
+    System.out.println("Wait for 30 seconds for Attestation to become invalid...");
+
+    //wait for 30 seconds
+    Thread.sleep(1000*30);
+
+    tar = contract.callVerifyTicketAttestation(useTicket.getDerEncoding());
+    assertFalse(tar.timeStampValid);
+
+    System.out.println("Ticket now invalid");
+
+    //Now run a test without the blockchain friendly timestamp:
+    att = HelperTest.makeUnsignedStandardAtt(subjectKeys.getPublic(), ATTESTATION_SECRET, MAIL, 15); //valid for 15 seconds
+    att.blockchainFriendly = false;
+    signed = new SignedIdentifierAttestation(att, attestorKeys);
+    ticket = new Ticket(MAIL, "6", TICKET_ID, TICKET_CLASS, ticketIssuerKeys, TICKET_SECRET); //using conferenceId 6 should pass (legacy test)
+    useTicket = new AttestedObject<>(ticket, signed, ATTESTATION_SECRET, TICKET_SECRET, UN, crypto);
+
+    //test should pass using legacy "6" value
+    tar = contract.callVerifyTicketAttestation(useTicket.getDerEncoding());
+    assertTrue(tar.timeStampValid);
+
+    System.out.println("Test with ConferenceID 6 passed: " + Numeric.toHexString(tar.conferenceId));
+
+    //now use legacy conferenceId 7, should fail
+    att = HelperTest.makeUnsignedStandardAtt(subjectKeys.getPublic(), ATTESTATION_SECRET, MAIL, 15); //valid for 15 seconds
+    att.blockchainFriendly = false;
+    signed = new SignedIdentifierAttestation(att, attestorKeys);
+    ticket = new Ticket(MAIL, "7", TICKET_ID, TICKET_CLASS, ticketIssuerKeys, TICKET_SECRET); //using conferenceId 6 should pass (legacy test)
+    useTicket = new AttestedObject<>(ticket, signed, ATTESTATION_SECRET, TICKET_SECRET, UN, crypto);
+
+    //test should fail:
+    tar = contract.callVerifyTicketAttestation(useTicket.getDerEncoding());
+    assertFalse(tar.timeStampValid);
+
```

Comment:
We should also have a tests that fail in case of
1. Wrong key used to sign the transaction.
2. Wrong Ticket issuer key is used
3. Wrong attestor key used

Reply:
We can test (2) and (3).

I agree we need a test for (1) too - I can add this into the HardHat .ts suite as it involves setting up a function which takes the attestation as payload, sending a transaction which calls that function and checks the result.